modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -1968,7 +1968,7 @@ void ONNXImporter::handleNode(const opencv_onnx::NodeProto& node_proto_)
                 layerParams.set("align_corners", interp_mode == "align_corners");
                 if (layerParams.get<String>("mode") == "linear")
                 {
-                    layerParams.set("mode", interp_mode == "pytorch_half_pixel" ?
+                    layerParams.set("mode", interp_mode == "pytorch_half_pixel" || interp_mode == "half_pixel" ?
                                             "opencv_linear" : "bilinear");
```

Comment:
Such modifications of DNN code requires proper regression testing.
Could you please prepare separate PR with regression tests for this change? You can take a look on #18841 as example (please target PR to 3.4 branch + opencv_extra PR with .onnx test models)

---

According to [ONNX spec of Resize](https://github.com/onnx/onnx/blob/master/docs/Operators.md#resize) current logic looks strange.
Resize-11 defines default mode for `coordinate_transformation_mode` attribute with `half_pixel` value. Which is not handled here. Version of ONNX opset is available through `.ir_version()` attribute.

Reply:
According to [ONNX Resize op Atributes](https://github.com/onnx/onnx/blob/master/docs/Changelog.md#attributes-182),you can see when `coordinate_transformation_mode ` == `half_pixel ` and `coordinate_transformation_mode ` == `pytorch_half_pixel ` actually have same way to handle data,because `length_resized` must greater than 1.But Previous logic `pytorch_half_pixel` goes to `opencv_linear`,but `half_pixel` goes to `bilinear`,resulting in wrong result as follow picture
![image](https://user-images.githubusercontent.com/19977378/123428981-fdb47580-d5f8-11eb-8a86-93ea93356fc1.png)
when i change code and recompile,the result of dnn looks good as follow picture
![image](https://user-images.githubusercontent.com/19977378/123429299-61d73980-d5f9-11eb-9751-8c2310dc1064.png)

Reply:
Main note here is about missing of proper regression testing for modification of DNN source code. There are no objections about this change.

Regression testing is necessary to automatically catch incorrect changes in the future.
Provided samples are not enough because they don't run automatically.